src/modules/Builder/features/PerformanceTasks/Flanker/Flanker.test.tsx

```diff
@@ -0,0 +1,724 @@
+    const ref = renderFlanker();
+
+    fireEvent.click(screen.getByTestId(`${mockedFlankerTestid}-stimulus-screen-add`));
+    fireEvent.click(screen.getByTestId(`${mockedFlankerTestid}-stimulus-screen-add`));
+
+    await waitFor(() => {
+      expandAllPanels();
+      expect(screen.queryByTestId(`${testId}-table`)).not.toBeInTheDocument();
+    });
+
+    ref.current.setValue('activities.0.items.2.config.stimulusTrials', [
+      {
+        image: 'image1',
+        text: 'text1',
+      },
+    ]);
+
+    await waitFor(() => {
+      expandAllPanels();
+      const table = screen.getByTestId(`${testId}-table`);
+      expect(table).toBeInTheDocument();
+      expect(table.querySelectorAll('tbody > tr')).toHaveLength(1);
+    });
+
+    ref.current.setValue('activities.0.items.2.config.stimulusTrials.1', {
+      image: 'image2',
+      text: 'text2',
+    });
+
+    await waitFor(() => {
+      expandAllPanels();
+      const table = screen.getByTestId(`${testId}-table`);
+      expect(table).toBeInTheDocument();
+      expect(table.querySelectorAll('tbody > tr')).toHaveLength(2);
+    });
+
+    const headCells = screen.getByTestId(`${testId}-table`).querySelectorAll('tr:last-child th');
+    expect(headCells).toHaveLength(4);
+    headCells.forEach((headCell, index) => {
+      expect(headCell).toHaveTextContent(`Block ${index + 1}`);
+    });
+
+    const bodyRows = screen.getByTestId(`${testId}-table`).querySelectorAll('tbody tr');
+    expect(bodyRows).toHaveLength(2);
+    const [firstRow, secondRow] = bodyRows;
+
+    expect(firstRow.querySelectorAll('td')).toHaveLength(4);
+    expect(secondRow.querySelectorAll('td')).toHaveLength(4);
+
+    firstRow.querySelectorAll('td').forEach((cell) => {
+      expect(cell).toHaveTextContent('text');
+    });
+    secondRow.querySelectorAll('td').forEach((cell) => {
+      expect(cell).toHaveTextContent('text');
+    });
+  });
+
+  describe('Validations', () => {
+    test.each`
+      testId                                                     | error                                   | description
+      ${`${mockedFlankerTestid}-buttons-0-text`}                 | ${'Button Name or Image is required'}   | ${'Left Button Name'}
+      ${`${mockedFlankerTestid}-buttons-1-text`}                 | ${'Button Name or Image is required'}   | ${'Right Button Name'}
+      ${`${mockedFlankerTestid}-stimulus-screen`}                | ${'Please fill in all required fields'} | ${'Stimulus Screen'}
+      ${`${mockedFlankerTestid}-practice-round-block-sequences`} | ${'Add stimulus screens first'}         | ${'Practice Round: Stimulus Screen'}
+      ${`${mockedFlankerTestid}-test-round-block-sequences`}     | ${'Add stimulus screens first'}         | ${'Test Round: Stimulus Screen'}
+    `('$description', async ({ testId, error }) => {
+      const ref = renderFlanker();
+
+      await ref.current.trigger('activities');
+
+      await waitFor(() => {
+        const section = screen.getByTestId(testId);
```

Comment:
you can get rid of `await waitFor()` wrapper, if you will use `find*` method:
`const section = await screen.findByTestId(testId);`